File: api_example.py
import requests
import base64

with open("static/images/index.jpeg", "rb") as img_file:
    img_str = base64.b64encode(img_file.read()).decode()

response = requests.post(
    "https://haddy-catdenfy.hf.space/run/predict",
    json={
        "data": [
            "data:image/png;base64,{}".format(img_str),
        ]
    },
).json()

results = response["data"][0]
print(results.get("label").replace("_", " "))
probs = results.get("confidences")
probs_dict = {list(p.values())[0].replace("_", " "): list(p.values())[1] for p in probs}
print(probs_dict)

File: api_example.js
#!/usr/bin/node

const fetch = require('node-fetch');
const fs = require('fs');

fs.readFile('static/images/index.jpeg', 'base64', (error, img) => {
  if (error) {
    console.log(error);
  } else {
    fetch('https://haddy-catdenfy.hf.space/run/predict', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        data: [
          'data:image/png;base64,' + img
        ]
      })
    })
      .then(r => r.json())
      .then(
        r => {
          const data = r.data[0];
          //   console.log(data.label.replace('_', ' '));
          //   console.log(data.confidences);
          const probsDict = {};
          for (let i = 0; i < data.confidences.length; i++) {
            probsDict[data.confidences[i].label.replace('_', ' ')] = data.confidences[i].confidence;
          }
          console.log('probability dictionary: ', probsDict);
          console.log('predicted class: ' + data.label);
        }
      );
  }
});
